### ISSUE TITLE
Build command fails on MacOS Sierra

### ISSUE THREAD

**Issue by aazon** [ansible/ansible-container#653]
##### ISSUE TYPE

 - Bug Report

##### container.yml 
Exact copy from [Getting Started](http://docs.ansible.com/ansible-container/getting_started.html) page. 

```
version: "2"
settings:
  conductor_base: centos:7
defaults:
  POSTGRES_USER: django
  POSTGRES_PASSWORD: sesame
  POSTGRES_DB: django
  DJANGO_PORT: 8080
services:
  django:
    from: centos:7
    roles:
      - django-gunicorn
    environment:
      DATABASE_URL: "pgsql://{{ POSTGRES_USER }}:{{ POSTGRES_PASSWORD }}@postgres:5432/{{ POSTGRES_DB }}"
      DJANGO_ROOT: '{{ DJANGO_ROOT }}'
      DJANGO_VENV: '{{ DJANGO_VENV }}'
    links:
    - postgres
    - postgres:postgresql
    ports:
    - '{{ DJANGO_PORT }}'
    working_dir: '{{ DJANGO_ROOT }}'
    user: '{{ DJANGO_USER }}'
    command: ['{{ DJANGO_VENV }}/bin/gunicorn', '-w', '2', '-b', '0.0.0.0:{{ DJANGO_PORT }}', 'project.wsgi:application']
    entrypoint: ['/usr/bin/dumb-init', '/usr/bin/entrypoint.sh']
    volumes:
      - 'static:{{ DJANGO_ROOT }}'
    dev_overrides:
      command: ['{{ DJANGO_VENV }}/bin/python', 'manage.py', 'runserver', '0.0.0.0:{{ DJANGO_PORT }}']
      volumes:
      - '${PWD}:{{ DJANGO_ROOT }}'
      - "static:/static"
      expose:
      - "{{ DJANGO_PORT }}"
      environment:
        DEBUG: "1"
  nginx:
    from: centos:7
    roles:
    - nginx
    ports:
    - '{{ DJANGO_PORT }}:8000'
    user: nginx
    links:
    - django
    command: ['/usr/bin/dumb-init', 'nginx', '-c', '/etc/nginx/nginx.conf']
    volumes:
    - 'static:{{ STATIC_ROOT }}'
    dev_overrides:
      ports: []
      command: /bin/false
      volumes: []
  postgres:
    from: postgres:9.6
    environment:
      POSTGRES_USER: "{{ POSTGRES_USER }}"
      POSTGRES_PASSWORD: "{{ POSTGRES_PASSWORD }}"
      POSTGRES_DB: "{{ POSTGRES_DB }}"
volumes:
  static:
    docker: {}
registries: {}
```

##### OS / ENVIRONMENT

```
Ansible Container, version 0.9.1
Darwin, Alexs-MacBook-Pro.local, 16.6.0, Darwin Kernel Version 16.6.0: Fri Apr 14 16:21:16 PDT 2017; root:xnu-3789.60.24~6/RELEASE_X86_64, x86_64
2.7.13 (default, Dec 18 2016, 07:03:39)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)] /usr/local/opt/python/bin/python2.7
{
  "ContainersPaused": 0,
  "Labels": null,
  "CgroupDriver": "cgroupfs",
  "ContainersRunning": 0,
  "ContainerdCommit": {
    "Expected": "cfb82a876ecc11b5ca0977d1733adbe58599088a",
    "ID": "cfb82a876ecc11b5ca0977d1733adbe58599088a"
  },
  "InitBinary": "docker-init",
  "NGoroutines": 29,
  "Swarm": {
    "ControlAvailable": false,
    "NodeID": "",
    "Error": "",
    "RemoteManagers": null,
    "LocalNodeState": "inactive",
    "NodeAddr": ""
  },
  "LoggingDriver": "json-file",
  "OSType": "linux",
  "HttpProxy": "",
  "Runtimes": {
    "runc": {
      "path": "docker-runc"
    }
  },
  "DriverStatus": [
    [
      "Root Dir",
      "/var/lib/docker/aufs"
    ],
    [
      "Backing Filesystem",
      "extfs"
    ],
    [
      "Dirs",
      "6"
    ],
    [
      "Dirperm1 Supported",
      "true"
    ]
  ],
  "OperatingSystem": "Alpine Linux v3.5",
  "Containers": 1,
  "HttpsProxy": "",
  "BridgeNfIp6tables": true,
  "MemTotal": 2095882240,
  "SecurityOptions": [
    "name=seccomp,profile=default"
  ],
  "Driver": "aufs",
  "IndexServerAddress": "https://index.docker.io/v1/",
  "ClusterStore": "",
  "InitCommit": {
    "Expected": "949e6fa",
    "ID": "949e6fa"
  },
  "Isolation": "",
  "SystemStatus": null,
  "OomKillDisable": true,
  "ClusterAdvertise": "",
  "SystemTime": "2017-07-15T09:33:33.250504097Z",
  "Name": "moby",
  "CPUSet": true,
  "RegistryConfig": {
    "AllowNondistributableArtifactsCIDRs": [],
    "Mirrors": [],
    "IndexConfigs": {
      "docker.io": {
        "Official": true,
        "Name": "docker.io",
        "Secure": true,
        "Mirrors": []
      }
    },
    "AllowNondistributableArtifactsHostnames": [],
    "InsecureRegistryCIDRs": [
      "127.0.0.0/8"
    ]
  },
  "DefaultRuntime": "runc",
  "ContainersStopped": 1,
  "NCPU": 4,
  "NFd": 17,
  "Architecture": "x86_64",
  "KernelMemory": true,
  "CpuCfsQuota": true,
  "Debug": true,
  "ID": "JZ5W:BZA2:UVVH:AXCX:R65R:7ZYG:AKUZ:ZFT5:MO6W:D7YI:GARJ:WGVQ",
  "IPv4Forwarding": true,
  "KernelVersion": "4.9.36-moby",
  "BridgeNfIptables": true,
  "NoProxy": "",
  "LiveRestoreEnabled": false,
  "ServerVersion": "17.06.0-ce",
  "CpuCfsPeriod": true,
  "ExperimentalBuild": true,
  "MemoryLimit": true,
  "SwapLimit": true,
  "Plugins": {
    "Volume": [
      "local"
    ],
    "Network": [
      "bridge",
      "host",
      "ipvlan",
      "macvlan",
      "null",
      "overlay"
    ],
    "Authorization": null,
    "Log": [
      "awslogs",
      "fluentd",
      "gcplogs",
      "gelf",
      "journald",
      "json-file",
      "logentries",
      "splunk",
      "syslog"
    ]
  },
  "Images": 5,
  "DockerRootDir": "/var/lib/docker",
  "NEventsListener": 1,
  "CPUShares": true,
  "RuncCommit": {
    "Expected": "2d41c047c83e09a6d61d464906feb2a2f3c52aa4",
    "ID": "2d41c047c83e09a6d61d464906feb2a2f3c52aa4"
  }
}
{
  "KernelVersion": "4.9.36-moby",
  "Arch": "amd64",
  "BuildTime": "2017-06-23T21:51:55.152028673+00:00",
  "ApiVersion": "1.30",
  "Version": "17.06.0-ce",
  "MinAPIVersion": "1.12",
  "GitCommit": "02c1d87",
  "Os": "linux",
  "Experimental": true,
  "GoVersion": "go1.8.3"
}
```
##### SUMMARY
I was following all steps described on [Getting Started](http://docs.ansible.com/ansible-container/getting_started.html) page. Execution of `ansible-container build` fails permanently.

##### STEPS TO REPRODUCE
- Create an empty directory and `cd` into it
- `ansible-container init`
- Copy the content of `container.yml` from [Getting Started](http://docs.ansible.com/ansible-container/getting_started.html) page.
- Run `ansible-container build`

##### EXPECTED RESULTS
No errors

##### ACTUAL RESULTS
```
$ ansible-container build
Building Docker Engine context...
Starting Docker build of Ansible Container Conductor image (please be patient)...
ERROR	Unknown exception
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/container/cli.py", line 268, in __call__
    getattr(core, u'hostcmd_{}'.format(args.subcommand))(**vars(args))
  File "/usr/local/lib/python2.7/site-packages/container/__init__.py", line 28, in __wrapped__
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/container/core.py", line 150, in hostcmd_build
    cache=kwargs['cache']
  File "/usr/local/lib/python2.7/site-packages/container/docker/engine.py", line 82, in __wrapped__
    return fn(self, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/container/__init__.py", line 28, in __wrapped__
    return fn(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/container/docker/engine.py", line 744, in build_conductor_image
    nocache=not cache)
  File "/usr/local/lib/python2.7/site-packages/docker/models/images.py", line 175, in build
    raise BuildError(chunk['error'])
BuildError: The command '/bin/sh -c python /get-pip.py &&     mkdir -p /etc/ansible/roles /_ansible/src &&     cd /usr/local/bin &&     tar -xz --strip-components=1 -f /tmp/docker.tgz' returned a non-zero code: 2
```
Here is a screenshot of error message from output with `--debug` flag
https://dl.dropboxusercontent.com/spa/hoang1aje3n09gn/hlp3-5we.png
![hlp3-5we](https://user-images.githubusercontent.com/378058/28238322-4a23dcf8-6951-11e7-94b1-d6ce09d35704.png)

Complete output is [here](https://gist.github.com/aazon/a07f464d85673541abd01d7bd7ecdc32)

Versions of `pip` packages:

```
$ pip list
DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.
ansible (2.3.1.0)
ansible-container (0.9.1)
ansible-lint (3.4.13)
appdirs (1.4.3)
asn1crypto (0.22.0)
backports.ssl-match-hostname (3.5.0.1)
bcrypt (3.1.3)
certifi (2017.4.17)
cffi (1.10.0)
chardet (3.0.4)
colorama (0.3.9)
cryptography (1.9)
docker (2.4.2)
docker-pycreds (0.2.1)
ecdsa (0.13)
enum34 (1.1.6)
idna (2.5)
ipaddress (1.0.18)
Jinja2 (2.9.6)
MarkupSafe (1.0)
mercurial (4.2.2)
packaging (16.8)
paramiko (2.2.1)
pip (9.0.1)
pyasn1 (0.2.3)
pycparser (2.18)
pycrypto (2.6.1)
Pygments (2.2.0)
PyNaCl (1.1.2)
pyparsing (2.2.0)
PyYAML (3.12)
requests (2.18.1)
ruamel.ordereddict (0.4.9)
ruamel.yaml (0.15.19)
setuptools (36.2.0)
six (1.10.0)
structlog (17.2.0)
urllib3 (1.21.1)
websocket-client (0.44.0)
wheel (0.29.0)
yq (2.2.0)
```

**Comment by marcusianlevine** [comment]
@aazon I was experiencing this as well with the latest Docker for Mac release on Sierra

As @chouseknecht describes in #627 this was addressed in #592 which was merged into `ansible:develop`

I was able to get past this error by building ansible-container from source as described [here](https://docs.ansible.com/ansible-container/installation.html#running-from-source)

**Comment by aazon** [comment]
@marcusianlevine Thanks a lot, this helped me as well!
